### PR TITLE
HHH-14603: Remove deprecated ClassLoaderService.fromConfigSettings me…

### PR DESCRIPTION
**Pull request by jrenaat**
…thod, as well as related AvailableSetting constants

    - correction for a few failing tests after refactoring

Signed-off-by: Jan Schatteman <jschatte@redhat.com>

### CODE REVIEW

**Comment by jrenaat** [comment]
Got the failing tests to work, but had to fall back on passing a classloader to the PersistenceXmlParser through the integration settings Map. Something about that just doesn't feel quite right, so I would imagine that there is prob. a better way to solve the issue.

File: hibernate-core/src/test/java/org/hibernate/orm/test/bootstrap/scanning/ScannerTest.java [hunk 1]
```diff
@@ -102,10 +104,5 @@ public void testCustomScanner() throws Exception {
 		emf = Persistence.createEntityManagerFactory( "defaultpar", integration );
 		assertTrue( CustomScanner.isUsed() );
 		emf.close();
-
-		CustomScanner.resetUsed();
-		emf = Persistence.createEntityManagerFactory( "defaultpar", null );
-		assertTrue( ! CustomScanner.isUsed() );
-		emf.close();
```
Comment (beikov) [inline hibernate-core/src/test/java/org/hibernate/orm/test/bootstrap/scanning/ScannerTest.java:109]: Was this removal on purpose?

**Review by beikov — COMMENTED** [review]
LGTM except for the removal of one test assertion.

File: hibernate-core/src/test/java/org/hibernate/orm/test/bootstrap/scanning/ScannerTest.java (hunk 1 above)
Comment (jrenaat) [reply hibernate-core/src/test/java/org/hibernate/orm/test/bootstrap/scanning/ScannerTest.java:109]: Indeed; this 'proposal' to fix the failures, is based on passing a ClassLoader to where it is needed through the integration settings Map, so having a test that nulls that settings map, sort of defeats that idea ...
This is one of the reasons why I mentioned that the solution 'doesn't feel quite right' ...

**Comment by sebersole** [comment]
Out of date
